### PR TITLE
Updated Independence Tests - Refactor 

### PR DESCRIPTION
**Pull request by ewengillies**

### CODE REVIEW

File: tigramite/independence_tests.py [hunk 1]
```diff
@@ -303,84 +108,156 @@ class CondIndTest(object):
...
+        """
+        mask_type : str, optional (default = None)
+            Must be in {'y','x','z','xy','xz','yz','xyz'}
+            Masking mode: Indicators for which variables in the dependence
+            measure I(X; Y | Z) the samples should be masked. If None, 'y' is
+            used, which excludes all time slices containing masked samples in Y.
+            Explained in [1]_.
+        """
+        if self.mask_type is not None:
+            mask_set = set(self.mask_type) - set(['x', 'y', 'z'])
```
Comment (ewengillies) [inline tigramite/independence_tests.py:199]: check that 'xyz' or 'xy' also works, i.e. more than one letter.

**Comment by ewengillies** [comment]
remove GPACE 

EDIT: done

**Comment by ewengillies** [comment]
Rename GaussProcReg

EDIT: done

**Comment by ewengillies** [comment]
Switch GaussProcReg from base class to instantiated member.
Have GaussProcReg inherit from nothing.

EDIT: done, but could still use some decoupling..

**Review by jakobrunge — APPROVED** [review]

File: tigramite/independence_tests.py (hunk 1 above)
Comment (ewengillies) [reply tigramite/independence_tests.py:199]: done.

**Comment by ewengillies** [comment]
okay @jakobrunge , please review this code, I've added the things we discussed.  I've also changed the tests a bit so they do not stall on Travis, all 426 tests run successfully on Python 2.7, 3.4, 3.5, and 3.6.  Here is the latest build: https://travis-ci.org/ewengillies/tigramite/builds/344454288

**Pull request merged**